crates/modelardb_manager/src/remote.rs

```diff
@@ -447,21 +447,23 @@ impl FlightService for FlightServiceHandler {
             let server_mode = ServerMode::from_str(mode).map_err(Status::invalid_argument)?;
             let node = Node::new(url.to_string(), server_mode.clone());
 
-            // Use the metadata manager to persist the node to the metadata Delta Lake.
+            // Use the cluster to register the node in memory. This returns an error if the node is
+            // already registered.
             self.context
-                .remote_data_folder
-                .metadata_manager
-                .save_node(&node)
+                .cluster
+                .write()
                 .await
+                .register_node(node.clone())
                 .map_err(|error| Status::internal(error.to_string()))?;
 
-            // Use the cluster to register the node in memory. Note that if this fails, the cluster
-            // and metadata Delta Lake will be out of sync until the manager is restarted.
+            // Use the metadata manager to persist the node to the metadata Delta Lake. Note that if
+            // this fails, the metadata Delta Lake and the cluster will be out of sync until the
+            // manager is restarted.
             self.context
-                .cluster
-                .write()
+                .remote_data_folder
+                .metadata_manager
+                .save_node(&node)
```

Comment:
Maybe change `save_node()` to take  ownership of `node` as it currently takes a reference but clones all members of `node`. By taking ownership of `node` it can reuse the members without of copying them and if the user only have a reference to `node` they can then copy instead of `save_node()` always making a copy.

Reply:
Fixed.